frontend/src/components/SearchBar.tsx

```diff
@@ -1,39 +1,48 @@
+import { zodResolver } from '@hookform/resolvers/zod';
 import { useForm } from 'react-hook-form';
 import { z } from 'zod';
 import { Form, FormControl, FormField, FormItem } from './ui/form';
 import { Search } from 'lucide-react';
 import { Input } from './ui/input';
-import { zodResolver } from '@hookform/resolvers/zod';
 import { Button } from './ui/button';
+import { useEffect } from 'react';
 
 const formSchema = z.object({
   searchQuery: z.string({
     required_error: 'Restaurant name is required',
   }),
 });
 
+export type SearchForm = z.infer<typeof formSchema>;
+
 type Props = {
   onSubmit: (formData: SearchForm) => void;
   placeHolder: string;
   onReset?: () => void;
+  searchQuery?: string;
 };
 
-export type SearchForm = z.infer<typeof formSchema>;
-
-const SearchBar = ({ onSubmit, placeHolder, onReset }: Props) => {
+const SearchBar = ({ onSubmit, onReset, placeHolder, searchQuery }: Props) => {
   const form = useForm<SearchForm>({
     resolver: zodResolver(formSchema),
+    defaultValues: {
+      searchQuery,
+    },
   });
 
-  function handleReset(): void {
+  useEffect(() => {
+    form.reset({ searchQuery });
+  }, [form, searchQuery]);
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/39#issuecomment-2408112738>#dc9d72</a><span> - 10/11/2024, 09:06 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The useEffect hook's dependency array is incomplete, which may lead to stale closures and unexpected behavior. A similar issue was also found in frontend/src/components/SearchBar.tsx (line 35).
</div>
<div id="fix">
<b>Fix:</b> Update the useEffect hook's dependency array to include all dependencies, including the form.reset function.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -33,6 +33,6 @@
    useEffect(() => {
      form.reset({ searchQuery });
 -  }, [form, searchQuery]);
 +  }, [form, searchQuery, form.reset]);
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged